app/src/org/commcare/adapters/ComboboxAdapter.java

```diff
@@ -0,0 +1,130 @@
+package org.commcare.adapters;
+
+import android.content.Context;
+import android.util.TypedValue;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.Filter;
+import android.widget.TextView;
+
+import org.commcare.dalvik.R;
+import org.javarosa.core.model.ComboboxFilterRule;
+import org.javarosa.core.model.FuzzyMatchFilterRule;
+import org.javarosa.core.model.MultiWordFilterRule;
+import org.javarosa.core.model.StandardFilterRule;
+
+import java.util.ArrayList;
+
+/**
+ * A custom adapter for use by a Combobox view. The filtering behavior of this adapter is determined
+ * by the implementation of choiceShouldBeShown() in its ComboboxFilterRule, which defines whether
+ * a given answer choice should be considered a match for the text entered by the user.
+ *
+ * @author Aliza Stone
+ */
+public class ComboboxAdapter extends ArrayAdapter<String> {
+
+    private float customTextSize;
+    protected final String[] allChoices;
+    protected String[] currentChoices;
+    protected ComboboxFilterRule filterRule;
+
+    public ComboboxAdapter(final Context context, final String[] objects, FilterType type) {
```

Comment:
Yep you're totally right, the structure of passing in the enum was left over from a version where I had many subclasses of `ComboboxAdapter` instead of the filter classes, and those did have a bunch of arguments to their constructor. Will change.